security/src/test/groovy/org/aggregatortech/dindora/security/authentication/providers/FileIDProviderTest.groovy

```diff
@@ -0,0 +1,192 @@
+package org.aggregatortech.dindora.security.authentication.providers
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException
+import org.aggregatortech.dindora.common.ServiceLocatorHelper
+import org.aggregatortech.dindora.common.io.system.SystemHelper
+import org.aggregatortech.dindora.common.test.BaseSpecification
+import org.aggregatortech.dindora.exceptions.IDStoreNotConfiguredException
+import org.aggregatortech.dindora.exceptions.MessageService
+import org.aggregatortech.dindora.security.authentication.token.AuthenticationCredentials
+import org.aggregatortech.dindora.security.authentication.token.UserNamePasswordCredentials
+
+class FileIDProviderTest extends BaseSpecification {
+
+
+    def "Test throws ID Store Not configured exception if configurations are not valid"() {
+        setup:
+
+
+        when :
+          // idstore location is not set
+        IDProvider idp = new FileIDProvider();
+        then :
+
+
+        try {
+            idp.configure();
+            assert false
+        }
+        catch (IDStoreNotConfiguredException ex) {
+            assert  ex instanceof IDStoreNotConfiguredException
+        }
+
+        when :
+        // idstore location is  empty
+       idp = new FileIDProvider();
+        SystemHelper helper = ServiceLocatorHelper.getServiceLocator().getService(SystemHelper.class);
```

Comment:
To make this test independent of any changes to SystemHelper, it will be better to use a mock of SystemHelper. This will also alleviate the need of writeProperty method.